transform/binary-plugin/examples/openshift/openshift.go

```diff
@@ -95,6 +245,90 @@ func isDefault(name string) bool {
 	return false
 }
 
+func UpdateBuildConfig(u unstructured.Unstructured, fields openshiftOptionalFields) (jsonpatch.Patch, error) {
+	jsonPatch := jsonpatch.Patch{}
+	js, err := u.MarshalJSON()
+	if err != nil {
+		return jsonPatch, err
+	}
+
+	buildConfig := &buildv1API.BuildConfig{}
+
+	err = json.Unmarshal(js, buildConfig)
+	if err != nil {
+		return jsonPatch, err
+	}
+	patch, err := updateSecret(buildConfig.Spec.Output.PushSecret, buildConfigOutputPushSecret, fields)
+	if err != nil {
+		return nil, err
+	}
+	jsonPatch = append(jsonPatch, patch...)
+	if buildConfig.Spec.Output.To != nil {
```

Comment:
@shawn-hurley So to follow-up on your "add comments to explain what's happening" -- would the following comment at this line be an example of what you're talking about (right above the if statement):
```
// Update output image reference if specified and a map entry is provided
```

Reply:
Maybe, I think that this one is understandable enough but if we have the time it could help someone who is unfamiliar with golang and/or k8s manifests? IDK thoughts?